### PR TITLE
[tune] Add `--columns` flag for CLI

### LINKED ISSUE #4523: [tune] Specify `--columns` for CLI
### Describe the problem
Right now it's not possible to specify the columns shown for `list-experiments` and `list-trials`. 

It's already implemented in `tune/commands.py`, but just needs to be exposed via `scripts.py`.

This should also support `result-columns`.

> This script is awesome. I was not able to find the tune CLI (even under ray in site_packages). However ran it manually via python <some_path>/python2.7/site-packages/ray/tune/scripts.py list-trials experiment2/ which produces experiment_tag besides other columns. However, is there a way to also show each hyperparameter separately and final metrics (i.e return value of _train()) after the trials terminate in separate columns? 

See: https://groups.google.com/d/msg/ray-dev/T5q7DkkGlpQ/ZzbOTIsVBwAJ

@andrewztan can you take a look at this?

cc @Nithanaroy

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?
Allow users to specify the columns shown for `list-experiments` and `list-trials`.

## Related issue number

Closes #4523 

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/tune/TuneClient.ipynb [hunk 1]
```diff
@@ -2,9 +2,51 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
```
Comment (richardliaw) [inline python/ray/tune/TuneClient.ipynb]: I think you committed some extra stuff?

File: python/ray/tune/scripts.py [hunk 2]
```diff
@@ -28,9 +28,26 @@ def cli():
...
     help="Select filter in the format '<column> <operator> <value>'.")
-def list_trials(experiment_path, sort, output, filter_op):
+@click.option(
+    "--columns",
+    default=None,
+    type=str,
+    help="Select columns to be displayed.")
+@click.option(
+    "--result-columns",
+    "result_columns",
```
Comment (richardliaw) [inline python/ray/tune/scripts.py]: supposed to have --?

**Comment by richardliaw** [comment]
BTW @andrewztan can you add a test for this?

File: python/ray/tune/scripts.py (hunk 2 above)
Comment (andrewztan) [reply python/ray/tune/scripts.py]: This is the convention used in Clicks 7.0 when the flag and corresponding variable name are different. Do we want to allow users to use `--result_columns` as a flag as well?

**Review by richardliaw — APPROVED** [review]
Awesome! Thanks for getting this in.

File: python/ray/tune/commands.py [hunk 3]
```diff
@@ -263,6 +267,8 @@ def list_experiments(project_path,
         sys.exit(0)
 
     info_df = pd.DataFrame(experiment_data_collection)
+    if not info_keys:
+        info_keys = DEFAULT_PROJECT_INFO_KEYS
```
Comment (richardliaw) [inline python/ray/tune/commands.py]: what was the difference between this and the previous version?

**Pull request merged**